tests/python/unittest/test_random.py

```diff
@@ -844,6 +836,48 @@ def testLarge(data, repeat):
     testLarge(mx.nd.arange(0, 100000).reshape((10, 10000)), 10)
     testLarge(mx.nd.arange(0, 100000).reshape((10000, 10)), 10)
 
+@with_seed()
+def test_randint():
+    dtypes = ['int32', 'int64']
+    for dtype in dtypes:
+        params = {
+            'low': -1,
+            'high': 3,
+            'shape' : (500, 500),
+            'dtype' : dtype,
+            'ctx' : mx.context.current_context()
+            }
+        mx.random.seed(128)
+        ret1 = mx.nd.random.randint(**params).asnumpy()
+        mx.random.seed(128)
+        ret2 = mx.nd.random.randint(**params).asnumpy()
+        assert same(ret1, ret2), \
+                "ndarray test: `%s` should give the same result with the same seed"
+
+@with_seed()
+def test_randint_extremes():
+    a = mx.nd.random.randint(dtype='int64', low=50000000, high=50000010, ctx=mx.context.current_context())
+    assert a>=50000000 and a<=50000010
+
+@with_seed()
+def test_randint_generator():
+    ctx = mx.context.current_context()
+    for dtype in ['int32', 'int64']:
+        for low, high in [(50000000, 50001000),(-50000000,-9900),(-500,199),(-2147483647,2147483647)]:
+            scale = high - low
+            buckets, probs = gen_buckets_probs_with_ppf(lambda x: ss.uniform.ppf(x, loc=low, scale=scale), 5)            
```

Comment:
Also this can be flaky since ppf calculation may be different for discrete distributions. Can we run this test a few 1000 times and also tweak some switches in verify_generator to avoid this.

Reply:
Addressed it for failing Python 2, GPU config 
added the test command on the PR description.